parity-clib/examples/cpp/main.cpp

```diff
@@ -14,172 +14,134 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
 
+#ifdef __clang__
+#pragma clang diagnostic error "-Wvexing-parse"
+#endif
 #include <chrono>
+#include <cstdint>
 #include <parity.h>
+#include <parity.hpp>
 #include <regex>
 #include <string>
 #include <thread>
+namespace {
+using namespace std::literals::string_literals;
+using parity::ethereum::ParityEthereum;
+using parity::ethereum::parity_subscription;
 
-void* parity_run(std::vector<const char*>);
-int parity_subscribe_to_websocket(void*);
-int parity_rpc_queries(void*);
+void parity_subscribe_to_websocket(ParityEthereum &ethereum);
+void parity_rpc_queries(ParityEthereum &);
+parity::ethereum::ParityEthereum parity_run(const std::vector<std::string> &);
 
 const int SUBSCRIPTION_ID_LEN = 18;
 const size_t TIMEOUT_ONE_MIN_AS_MILLIS = 60 * 1000;
 const unsigned int CALLBACK_RPC = 1;
 const unsigned int CALLBACK_WS = 2;
```

Comment:
I know that this code does not technically belong to your PR but please convert them to `constexpr` because we can - and also make use of proper sized int types as you did in the rest of the code. :)